ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -1022,4 +1046,53 @@ public static String stripSlashAndExtensionFromId(String blobIdWithExtension) {
     int endIndex = extensionIndex != -1 ? extensionIndex : blobIdWithExtension.length();
     return blobIdWithExtension.substring(startIndex, endIndex);
   }
+
+  /**
+   * Class to encode decode kv header values.
+   */
+  static class KVHeaderValueEncoderDecoder {
+    private static String DELIM = "; ";
+    private static String KV_SEPERATOR = "=";
+
+    /**
+     * Encode a map of {@link String}s as http header value.
+     * @param map {@link Map} of {@link String}s to encode.
+     * @return encoded http header value.
+     */
+    static String encodeKVHeaderValue(Map<String, String> map) {
+      Objects.requireNonNull(map);
+      String value = "";
+      for (Map.Entry<String, String> entry : map.entrySet()) {
+        value += entry.getKey() + KV_SEPERATOR + entry.getValue() + DELIM;
+      }
+      return value;
+    }
+
+    /**
+     * Decode a kv header value.
+     * @param value {@link String} containing the encoded kv values.
+     * @return decoded kv {@link Map}.
+     */
+    static Map<String, String> decodeKVHeaderValue(String value) {
```

Comment:
Can you make this class and method public so it can be called outside the package?